src/Agent.Listener/SelfUpdater.cs

```diff
@@ -36,14 +36,42 @@ public override void Initialize(IHostContext hostContext)
             _terminal = hostContext.GetService<ITerminal>();
             _agentServer = HostContext.GetService<IAgentServer>();
             var configStore = HostContext.GetService<IConfigurationStore>();
-            var settings = configStore.GetSettings();
-            _poolId = settings.PoolId;
-            _agentId = settings.AgentId;
+            _agentSettings = configStore.GetSettings();
         }
 
         public async Task<bool> SelfUpdate(AgentRefreshMessage updateMessage, IJobDispatcher jobDispatcher, bool restartInteractiveAgent, CancellationToken token)
         {
-            if (!await UpdateNeeded(updateMessage.TargetVersion, token))
+            var extensionManager = HostContext.GetService<IExtensionManager>();
```

Comment:
@TingluoHuang  Does this get called normally? Will it at least get called if I have a task that requires a higher versioned agent than is currently running?